src/gcode-streamer_test.cc

```diff
@@ -0,0 +1,208 @@
+/* -*- mode: c++; c-basic-offset: 2; indent-tabs-mode: nil; -*-
+ * Test for pru motion queue.
+ *
+ * We simulate the pru hardware and test if the absolute position is correctly
+ * retrieved.
+ *
+ */
+#include "motion-queue.h"
+
+#include <stdio.h>
+#include <string.h>
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include "common/fd-mux.h"
+#include "common/logging.h"
+#include "gcode-parser/gcode-parser.h"
+#include "gcode-streamer.h"
+
+class MockStream {
+public:
+
+  MockStream() : fd_{-1, -1} {}
+  ~MockStream() { ClientDisconnect(); }
+
+  int ClientConnect() {
+    ClientDisconnect();
+    if (pipe(fd_) < 0) {
+      perror("pipe()");
+      exit(1);
+    }
+    return fd_[0];
+  }
+
+  int NewData(const char *data) {
+    // Feed data in the pipe
+    return write(fd_[1], data, strlen(data));
+  }
+
+  void ClientDisconnect() { close(fd_[1]); }
+
+private:
+  int fd_[2];
+};
+
+class StreamTester : public GCodeParser::EventReceiver {
+public:
+  StreamTester() {
+    GCodeParser::Config config;
+    parser_ = new GCodeParser(config, this, false);
+    streamer_ = new GCodeStreamer(&event_server_, parser_, this);
+    stream_mock_ = NULL;
+  }
+
+  ~StreamTester() {
+    delete parser_;
+    delete streamer_;
+    delete stream_mock_;
+  }
+
+  bool OpenStream() {
+    assert(stream_mock_ == NULL);
+    stream_mock_ = new MockStream();
+    return streamer_->ConnectStream(stream_mock_->ClientConnect(), NULL);
+  }
+
+  void CloseStream() {
+    stream_mock_->ClientDisconnect();
+    delete stream_mock_;
+    stream_mock_ = NULL;
+  }
+  void ReceiveString(const char *line) {
```

Comment:
This `ReceiveString()` is very hard to understand, as it actually describes the opposite what it does.
It should probably be named `SendString()`.

Reply:
(solved)